### ISSUE TITLE
Annotation on a CronJob doesn't work?

### ISSUE THREAD

**Issue by apenney** [jtblin/kube2iam#65]
Hi,

We're using kube2iam and it's been working great, but we just introduced our first CronJob into the cluster.  This creates a pod without the annotation.  I've been trying to trace through the code on both sides, but I'm pretty clueless.  Is this something that can be fixed on the kube2iam side to magically work?

**Comment by jtblin** [comment]
@apenney have you checked #49? Seems like the same issue that you can solve by adding annotations to the pod template in your job spec.

Could you confirm if that works please and I'll update the readme accordingly?

**Comment by hjacobs** [comment]
@apenney I know for sure that `kube2iam` works with CronJob as we are making heavy use of it (Kubernetes 1.6.2), here an example manifest (double curly braces are our own templating):

```yaml
apiVersion: batch/v2alpha1
kind: CronJob
metadata:
  name: "{{ application }}"
  labels:
    application: "{{ application }}"
    version: "{{ version }}"
    release: "{{ release }}"
    stage: production
spec:
  schedule: "00 11 * * 2"
  concurrencyPolicy: Forbid
  startingDeadlineSeconds: 3600
  jobTemplate:
    spec:
      template:
        metadata:
          labels:
            application: "{{ application }}"
            version: "{{ version }}"
            release: "{{ release }}"
            stage: production
          annotations:
            iam.amazonaws.com/role: app-{{ application }}-1
        spec:
          restartPolicy: OnFailure
          containers:
              ...
```

If you are using `CronJob` or jobs in general, please also have a look at https://kubernetes-on-aws.readthedocs.io/en/latest/admin-guide/kubernetes-in-production.html#jobs which links a small script to clean up jobs/pods (you will pollute your API server otherwise which becomes dead slow).

**Comment by apenney** [comment]
Thanks to you both!  I believe the issue was we had a stuck job (had to delete with --cascade=false in the end) and it was causing the pod to be regenerated constantly with the annotation in the wrong place.

We actually fixed it, redeployed, but it was ignoring us and continuing to spin up the broken pod.  I just launched this fresh after reading @hjacobs comment and it seems to be working OK and I can see the annotation when I get the pod.

Thanks to you both!  Looking forward to getting to kubes 1.6 so this stuck job thing goes away (I saw it was fixed upstream!)
